### ISSUE TITLE
Nullability mismatch reported for generic return type of implementation of unannotated method

### ISSUE THREAD

**Issue by cston** [dotnet/roslyn#29675]
A warning is reported for `IEnumerable<T>.GetEnumerator()`:
```c#
using System.Collections;
using System.Collections.Generic;
using System.Runtime.CompilerServices;
[module: NonNullTypes]
class Enumerable<T> : IEnumerable<T>
{
    IEnumerator<T> IEnumerable<T>.GetEnumerator() => throw null; // warning
    IEnumerator IEnumerable.GetEnumerator() => throw null;
}
```
```
(7,35): warning CS8616: Nullability of reference types in return type doesn't match
    implemented member 'IEnumerator<T> IEnumerable<T>.GetEnumerator()'.
```

**Comment by cston** [comment]
Fixed in #30913?

**Comment by DavidArno** [comment]
Did #30913 make it into VS2019 preview 1? If it did, this problem still persists as I'm getting the same warning testing C# 8 beta there.

**Comment by alrz** [comment]
@DavidArno The merged PR is targeting preview 2 branch.

**Comment by DavidArno** [comment]
@alrz, thanks. I couldn't work that out for myself, so glad you were on hand to work it out for me! 😁
